ARGO/src/Controller.cpp

```diff
@@ -0,0 +1,272 @@
+#endif // _DEBUG
+	AxisState currentState = AxisState::NotMoved;
+
+	if (AxisType::LeftThumbStick == t_axisType && m_current.LeftThumbStick != m_previous.LeftThumbStick)
+	{
+#ifdef _DEBUG 
+		debugString += "Left Thumb Stick Moved, Position: X: " + std::to_string(m_current.LeftThumbStick.x) + " ,Y: " + std::to_string(m_current.LeftThumbStick.y);
+#endif // _DEBUG
+		currentState = AxisState::Moved;
+	} 
+	else if (AxisType::RightThumbStick == t_axisType && m_current.RightThumbStick != m_previous.RightThumbStick)
+	{
+#ifdef _DEBUG
+		debugString += "Right Thumb Stick Moved, Position: X: " + std::to_string(m_current.RightThumbStick.x) + " ,Y: " + std::to_string(m_current.RightThumbStick.y);
+#endif // _DEBUG
+		currentState = AxisState::Moved;
+	}
+	else if (AxisType::LeftTrigger == t_axisType && m_current.LTrigger != m_previous.LTrigger)
+	{
+#ifdef _DEBUG
+		debugString += "Left Trigger Moved, Value: " + std::to_string(m_current.LTrigger);
+#endif // _DEBUG
+		currentState = AxisState::Moved;
+	}
+	else if(AxisType::RightTrigger == t_axisType && m_current.RTrigger != m_previous.RTrigger)
+	{
+#ifdef _DEBUG
+		debugString += "Right Trigger Moved, Value: " + std::to_string(m_current.RTrigger);
+#endif // _DEBUG
+		currentState = AxisState::Moved;
+	}
+
+#ifdef _DEBUG
+	if (AxisState::NotMoved != currentState)
+	{
+		std::cout << debugString << std::endl;
+	}
+#endif // _DEBUG
+
+	return currentState;
+}
+
+SDL_GameController* Controller::getSDLController()
+{
+	return m_controller;
+}
+
+void Controller::activateRumble(RumbleStrength t_strength, RumbleLength t_length)
+{
+	m_rumble.activateRumble(t_strength, t_length);
+}
+
+void Controller::setRumbleStrength(float t_newStrength)
+{
+	m_rumble.setStrength(t_newStrength);
+}
+
+void Controller::setRumbleTime(float t_time)
+{
+	m_rumble.setTime(t_time);
+}
+
+float Controller::getRumbleStrength()
+{
+	return m_rumble.getStrength();
+}
+
+float Controller::getRumbleTime()
+{
+	return m_rumble.getTime();
+}
+
+/// <summary>
+/// Returns the name of the button that's type is passed to the function
+/// </summary>
+/// <param name="t_buttonType">type of button you want the name of</param>
+/// <returns>name of button</returns>
+std::string Controller::getButtonName(ButtonType t_buttonType)
+{
+	switch (t_buttonType)
+	{
+	case ButtonType::A:
+		return "A";
+	case ButtonType::B:
+		return "B";
+	case ButtonType::X:
+		return "X";
+	case ButtonType::Y:
+		return "Y";
+	case ButtonType::RB:
+		return "RB";
+	case ButtonType::LB:
+		return "LB";
+	case ButtonType::LeftThumbStickClick:
+		return "LEFT STICK CLICK";
+	case ButtonType::RightThumbStickClick:
+		return "RIGHT STICK CLICK";
+	case ButtonType::DpadUp:
+		return "DPAD UP";
+	case ButtonType::DpadDown:
+		return "DPAD DOWN";
+	case ButtonType::DpadRight:
+		return "DPAD RIGHT";
+	case ButtonType::DpadLeft:
+		return "DPAD LEFT";
+	case ButtonType::Start:
+		return "START";
+	case ButtonType::Back:
+		return "BACK";
```

Comment:
All of these are magic strings, wouldn't it make more sense to have these consts